src/behaviours/preferences/navigation-to-extension-specific-preferences.test.tsx

```diff
@@ -87,14 +144,143 @@ describe("preferences - navigation to extension specific preferences", () => {
+        const actual = rendered.getByTestId("tab-link-for-extension-registered-tab-page-id-nav-item-metrics-extension-tab");
+
+        expect(actual).toMatchSnapshot();
+      });
+
+      it("does not show custom settings block", () => {
+        const actual = rendered.queryByTestId("extension-settings");
+
+        expect(actual).not.toBeInTheDocument();
+      });
+
+      describe("when navigating to specific extension tab", () => {
+        beforeEach(() => {
+          applicationBuilder.preferences.navigation.click("extension-registered-tab-page-id-nav-item-metrics-extension-tab");
+        });
+        it("renders", () => {
+          expect(rendered.container).toMatchSnapshot();
+        });
+        it("shows related preferences for this tab", () => {
+          const actual = rendered.getByTestId("metrics-preference-item-hint");
+
+          expect(actual).toBeInTheDocument();
+        });
+        it("does not show unrelated preferences for this tab", () => {
+          const actual = rendered.queryByTestId("survey-preference-item-hint");
+
+          expect(actual).not.toBeInTheDocument();
+        });
+      });
+    });
+
+    describe("given extension with few registered tabs", () => {
+      beforeEach(async () => {
+        const getRendererExtensionFake = getRendererExtensionFakeFor(applicationBuilder);
+        const extension = getRendererExtensionFake(extensionStubWithWithRegisteredTabs);
+
+        await applicationBuilder.addExtensions(extension);
+      });
+
+      it("shows each of registered tabs in general area", () => {
+        const helloTab = rendered.getByTestId("tab-link-for-extension-hello-world-tab-page-id-nav-item-hello-extension-tab");
+        const logsTab = rendered.getByTestId("tab-link-for-extension-hello-world-tab-page-id-nav-item-logs-extension-tab");
+
+        expect(helloTab).toBeInTheDocument();
+        expect(logsTab).toBeInTheDocument();
+      });
+    });
+
+    describe("given extensions with tabs having same id", () => {
+      beforeEach(async () => {
+        const getRendererExtensionFake = getRendererExtensionFakeFor(applicationBuilder);
+        const extension = getRendererExtensionFake(extensionStubWithWithRegisteredTab);
+        const otherExtension = getRendererExtensionFake(extensionStubWithWithSameRegisteredTab);
+
+        await applicationBuilder.addExtensions(extension, otherExtension);
+      });
+
+      it("shows tab from the first extension", () => {
+        const actual = rendered.getByTestId("tab-link-for-extension-registered-tab-page-id-nav-item-metrics-extension-tab");
+
+        expect(actual).toBeInTheDocument();
+      });
+
+      it("shows tab from the second extension", () => {
+        const actual = rendered.getByTestId("tab-link-for-extension-duplicated-tab-page-id-nav-item-metrics-extension-tab");
+
+        expect(actual).toBeInTheDocument();
+      });
+
+      describe("when navigating to first extension tab", () => {
```

Comment:
I think that we should also specify the behaviour when navigating to a specific settings tab from outside the settings page, especially when there are multiple extensions that register the same ID.

Furthermore, I would like to see some tests about two extensions where extensions A registers a new tab and extension B uses that tab.

Reply:
Add some external test cases navigating directly to extension specific tabs. 

> Furthermore, I would like to see some tests about two extensions where extensions A registers a new tab and extension B uses that tab.

Added such test case too, and tabs registered in different extension don't affect each other, because specific tab route "located" inside extension route, for example
`path: "/preferences/extension/:extensionId/:tabId?"`